exercises/tree-building/Example.fs

```diff
@@ -5,13 +5,16 @@ type Tree =
     | Branch of int * Tree list
     | Leaf of int
 
+let rootNodeRecordId = 0
+
 let addOrAppend key value map =
     let list = defaultArg (Map.tryFind key map) []
     Map.add key (list @ [value]) map
 
 let invalidNode previous x = 
-    x.ParentId >= x.RecordId || 
-    x.RecordId <> previous + 1
+    match x.RecordId with
+    | 0 -> x.ParentId <> rootNodeRecordId
```

Comment:
Nope, because that's not how F# pattern matching works. It will not match on the value, but on the variable with that name. Therefore, it would then match on any input. For an example, see [this fiddle](https://dotnetfiddle.net/GA3oIn).